compiler/damlc/daml-stdlib-src/DA/Stack.daml

```diff
@@ -0,0 +1,90 @@
+-- Copyright (c) 2020 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+-- SPDX-License-Identifier: Apache-2.0
+
+{-# LANGUAGE ImplicitParams #-}
+module DA.Stack
+  ( CallStack
+  , HasCallStack
+  , SrcLoc(..)
+  , callStack
+  , getCallStack
+  , prettyCallStack
+  ) where
+
+import Prelude
+import DA.Text
+import GHC.Stack.Types hiding (SrcLoc(..))
+import qualified GHC.Stack.Types as GHC
+
+-- | Request a CallStack.
+--
+-- Calls to functions with this constraint will be added to the callstack.
+-- You can get access to the current call stack with `callStack`.
+--
+-- Note that if the call stack is reset if any function in between does not
+-- have a `HasCallStack` constraint.
+type HasCallStack = (?callStack : CallStack)
+
+-- Pretty-print a `CallStack`.
+prettyCallStack : CallStack -> Text
+prettyCallStack = intercalate "\n" . prettyCallStackLines
+
+prettyCallStackLines : CallStack -> [Text]
+prettyCallStackLines cs = case getCallStack cs of
+  []  -> []
+  stk -> "CallStack (from HasCallStack):"
+       :: map (("  " <>) . prettyCallSite) stk
+  where
+    prettyCallSite (f, loc) = f <> ", called at " <> prettySrcLoc loc
+
+prettySrcLoc : SrcLoc -> Text
+prettySrcLoc SrcLoc {..}
+  = foldr (<>) ""
```

Comment:
Could we use `DA.Text.implode` here?

Reply:
Good point, I just copied the GHC source code since I’m lazy. I’ll change it. Although now I wonder why GHC doesn’t use `concat` :thinking: